src/common/transformations/src/transformations/mlir/convert.cpp

```diff
@@ -553,11 +561,15 @@ template <typename TargetOp>
 struct ConvertBinary {
     void operator()(ConversionContext& context, std::shared_ptr<ov::Node> node) {
         auto loc = createLocation(context.context, node);
+        auto& builder = context.builder();
         // TODO: Support broadcasts
         const auto inputs = context.getInputs(node);
-        auto op = context.builder().create<TargetOp>(loc,
-                                                     mlir::ValueRange{inputs[0], inputs[1]},
-                                                     /* FIXME: Use linalg.fill or tensor.empty */ mlir::ValueRange{inputs[0]});
+        auto outType = cast<mlir::ShapedType>(inputs[0].getType());
+        // Named binary ops directly overwrite data in `outs` buffer so, there is no need to provide non-empty
+        // destination at the tensor-level.
+        // Use `tensor.empty` to avoid temporary buffer allocation and memcpy after bufferization.
+        auto empty = builder.create<mlir::tensor::EmptyOp>(loc, outType.getShape(), outType.getElementType());
```

Comment:
Does it work for dynamic shapes? When I experimented with something similar I didn't managed to create an output placeholder in such a simple way because it required materialization of dynamic dimensions as values in the MLIR program.

Reply:
Good point, I'm used to having everything static.
For dynamic shape, we need to provide dynamic dimensions' sizes which can be done at runtime with `tensor.dim` that gets cleaned up for static shapes. I'll adjust the code.

Reply:
It would be nice, thank you. I would like to keep code capable to work with dynamic shapes from the day zero to better understand the problem.

I tried to capture the dynamic broadcast semantics as well (as this Add and other element-wise ops in OV naturally have it) immediately at my first attempt of the lowering but it required to produce more essential logic that deals with dimensions materialized in the mlir module, so I postponed it. But basically I saw the code that we need in TOSA to Linalg lowering code in MLIR tree. Now we can skip this dynamic broadcast part.